core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -144,15 +144,14 @@ private function addCspHandler(ResponseContext $context, PageModel $pageModel):
 
             try {
                 $reportUri = $this->urlGenerator->generate(CspReporterController::class, ['page' => $pageModel->id], UrlGeneratorInterface::ABSOLUTE_URL);
-                $directives->setDirective('report-uri', $reportUri);
+                $cspHandler->getDirectives()->setDirective('report-uri', $reportUri);
```

Comment:
Why did you move this into the `try` block?

Reply:
I did not move anything in the try block? It is where it was before, only the `$directives` does not exist anymore.